x/dex/exchange/limit_order.go

```diff
@@ -14,41 +15,35 @@ func MatchLimitOrders(
 	settlements := []*types.SettlementEntry{}
 	totalExecuted, totalPrice := sdk.ZeroDec(), sdk.ZeroDec()
 	minPrice, maxPrice := sdk.NewDecFromInt(sdk.NewIntFromUint64(math.MaxInt64)), sdk.OneDec().Neg()
-	longPtr, shortPtr := len(orderbook.Longs.Entries)-1, 0
 
-	for longPtr >= 0 && shortPtr < len(orderbook.Shorts.Entries) && orderbook.Longs.Entries[longPtr].GetPrice().GTE(orderbook.Shorts.Entries[shortPtr].GetPrice()) {
+	for longEntry, shortEntry := orderbook.Longs.Next(ctx), orderbook.Shorts.Next(ctx); longEntry != nil && shortEntry != nil && longEntry.GetPrice().GTE(shortEntry.GetPrice()); longEntry, shortEntry = orderbook.Longs.Next(ctx), orderbook.Shorts.Next(ctx) {
 		var executed sdk.Dec
-		if orderbook.Longs.Entries[longPtr].GetEntry().Quantity.LT(orderbook.Shorts.Entries[shortPtr].GetEntry().Quantity) {
-			executed = orderbook.Longs.Entries[longPtr].GetEntry().Quantity
+		if longEntry.GetOrderEntry().Quantity.LT(shortEntry.GetOrderEntry().Quantity) {
+			executed = longEntry.GetOrderEntry().Quantity
 		} else {
-			executed = orderbook.Shorts.Entries[shortPtr].GetEntry().Quantity
+			executed = shortEntry.GetOrderEntry().Quantity
 		}
 		totalExecuted = totalExecuted.Add(executed).Add(executed)
 		totalPrice = totalPrice.Add(
 			executed.Mul(
-				orderbook.Longs.Entries[longPtr].GetPrice().Add(orderbook.Shorts.Entries[shortPtr].GetPrice()),
+				longEntry.GetPrice().Add(shortEntry.GetPrice()),
 			),
 		)
-		minPrice = sdk.MinDec(minPrice, orderbook.Longs.Entries[longPtr].GetPrice())
-		maxPrice = sdk.MaxDec(maxPrice, orderbook.Longs.Entries[longPtr].GetPrice())
+		minPrice = sdk.MinDec(minPrice, longEntry.GetPrice()) // tony: need to revisit whether it's intended to use long entry price here. Keeping it as-is so that test expectations are met
```

Comment:
@LCyson is it intentional here to use long entry price to get minPrice? not sure how minPrice is used with TriggeredOrder

Reply:
hm yea likely an issue here, minPrice should be the lowest short

Reply:
we can leave it as it is for now, we already remove support for TRIGGER ORDER correct